### DISCUSSION TITLE
How do I set the domain name resolution of the mailbox service?

### DISCUSSION THREAD

**Discussion by xiaofanshifu** [AdguardTeam/AdGuardHome#3874] [Q&A]
How do I set the domain name resolution of the mailbox service? 

I notice that in DNS rewriting, only two setting options are provided. One is domain and the other is answer. 
The types I need are a, MX, TXT, CNAME and priority.

example: a.com

| name | type  | value          | priority |
| ---- | ----- | -------------- | -------- |
| mail | A     | 192.168.3.12   |          |
| @    | MX    | mail.a.com     | 10       |
| @    | TXT   | v=spf1 mx -all |          |
| imap | CNAME | mail.a.com     |          |

Who can help me!

**Comment by xiaofanshifu** [comment]
my adg version: v0.107.0-b.15

Can I use custom filtering rules?
![image](https://user-images.githubusercontent.com/57950591/143185986-1024e771-fd1b-4e9f-bf5c-795e685f8fdb.png)

**Comment by xiaofanshifu** [comment]
I wrote a TXT record and encountered a question
![image](https://user-images.githubusercontent.com/57950591/143187403-53e71f69-1e08-4ef6-a2bd-2b2403b66512.png)
Does this ";" need special treatment? For example, use escape characters `\`

@ameshkov

**Reply by ameshkov** [reply]
That's a good question.

@ainar-g I don't really remember if we use escape characters in `$dnsrewrite`. In any case, this question needs to be addressed in the Wiki: https://github.com/AdguardTeam/AdGuardHome/wiki/Hosts-Blocklists#dnsrewrite

**Reply by ainar-g** [reply]
@ameshkov, currently, we simply interpret everything past the second semicolon as the value, so that semicolon doesn't require any special treatment.

If we want to retain that behaviour in the future, it's better to file a documentation issue to make sure we won't forget about it.

**Reply by xiaofanshifu** [reply]
@ameshkov @ainar-g Thank you, I've solved the problem.
However, I sincerely hope to use the graphical interface configuration, like DNS rewriting.

**Reply by ameshkov** [reply]
> we simply interpret everything past the second semicolon as the value, so that semicolon doesn't require any special treatment

This might be a problem since we cannot specify multiple modifiers this way:
`||example.org^$dnsrewrite=example.org,client=client`

**Reply by ainar-g** [reply]
@ameshkov, well that would be a breaking change, so even if you want to change it, I'd recommend only doing it in the v0.108 cycle.
